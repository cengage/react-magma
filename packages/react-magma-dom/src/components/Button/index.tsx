import * as React from 'react';
import { StyledButton } from '../StyledButton';
import { useIsInverse } from '../../inverse';
import { ButtonGroupContext } from '../ButtonGroup';
import { resolveProps } from '../../utils';

export enum ButtonVariant {
  solid = 'solid', //default
  link = 'link',
}

export enum ButtonColor {
  primary = 'primary', //default
  secondary = 'secondary',
  danger = 'danger',
  marketing = 'marketing',
  subtle = 'subtle',
}

export enum ButtonShape {
  fill = 'fill', //default
  leftCap = 'leftCap',
  rightCap = 'rightCap',
  round = 'round', // default for icon only buttons
}

export enum ButtonSize {
  large = 'large',
  medium = 'medium', //default
  small = 'small',
}

export enum ButtonTextTransform {
  uppercase = 'uppercase', //default
  none = 'none',
}

export enum ButtonType {
  button = 'button',
  submit = 'submit',
  reset = 'reset',
}

export interface ButtonStyles {
  /**
   * The color of the button, indicating its function in the UI
   * @default ButtonColor.primary
   */
  color?: ButtonColor;
  /**
   * Set the button to display full-width.
   * @default false
   */
  isFullWidth?: boolean;
  isInverse?: boolean;
  /**
   * Defines the border radius
   * @default ButtonShape.fill
   */
  shape?: ButtonShape;
  /**
   *  Set the button to a loading state
   * @default false
   */
  isLoading?: boolean;
  /**
   * The relative size of the button
   * @default ButtonSize.medium
   */
  size?: ButtonSize;
  /**
   * Determines whether the button appears in all-caps
   * @default ButtonTextTransform.uppercase
   */
  textTransform?: ButtonTextTransform;
  /**
   * The variant of the button
   * @default ButtonVariant.solid
   */
  variant?: ButtonVariant;
}

/**
 * @children required
 */
interface BaseButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  /**
   * @internal
   */
  testId?: string;
  /**
   * @internal
   */
  theme?: any;
  /**
   * The type attribute of the button
   * ButtonType.button
   */
  type?: ButtonType;
}

export type ButtonProps = BaseButtonProps & ButtonStyles;

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => {
    const contextProps = React.useContext(ButtonGroupContext);
    const resolvedProps = resolveProps(contextProps, props);
    const { color, shape, size, testId, textTransform, variant, ...rest } =
      resolvedProps;
    const isInverse = useIsInverse(resolvedProps.isInverse);

    return (
      <StyledButton
        {...rest}
        ref={ref}
        color={color || ButtonColor.primary}
        isInverse={isInverse}
        shape={shape || ButtonShape.fill}
        size={size || ButtonSize.medium}
        testId={testId}
        textTransform={textTransform || ButtonTextTransform.uppercase}
        variant={variant || ButtonVariant.solid}
      >
        {resolvedProps.children}
      </StyledButton>
    );
  }
);
