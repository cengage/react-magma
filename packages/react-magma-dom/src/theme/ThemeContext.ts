import * as React from 'react';

import { magma } from './magma';

export const ThemeContext = React.createContext(
  magma // default value
);
