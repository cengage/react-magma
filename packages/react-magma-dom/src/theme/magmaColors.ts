export enum MagmaColors {
  primary = 'primary',
  secondary = 'secondary',
  tertiary = 'tertiary',
  neutral = 'neutral',
  info = 'info',
  danger = 'danger',
  warning = 'warning',
  success = 'success',
}
