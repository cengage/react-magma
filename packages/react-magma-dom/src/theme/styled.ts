import styled from '@emotion/styled-base';
import { CreateStyled } from '@emotion/styled/types';
import { ThemeInterface } from './magma';

export default styled as CreateStyled<ThemeInterface>;
