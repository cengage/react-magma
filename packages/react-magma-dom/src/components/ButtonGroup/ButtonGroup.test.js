import React from 'react';
import { axe } from '../../../axe-helper';
import { ButtonGroup, ButtonGroupOrientation, ButtonGroupAlignment } from '.';
import { Button, ButtonColor, ButtonSize, ButtonVariant } from '../Button';
import { render } from '@testing-library/react';
import { magma } from '../../theme/magma';

const testId = 'test-id';

describe('ButtonGroup', () => {
  it('should find element by testId', () => {
    const { getByTestId } = render(
      <ButtonGroup testId={testId}>
        <Button>1</Button>
        <Button>2</Button>
        <Button>3</Button>
      </ButtonGroup>
    );

    expect(getByTestId(testId)).toBeInTheDocument();
  });

  it('does not violate accessibility standards', () => {
    const { container } = render(
      <ButtonGroup>
        <Button>1</Button>
        <Button>2</Button>
        <Button>3</Button>
      </ButtonGroup>
    );

    return axe(container.innerHTML).then(result => {
      return expect(result).toHaveNoViolations();
    });
  });

  describe('Horizontal', () => {
    describe('Alignment', () => {
      it('Default: aligns the buttons to the left', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.horizontal}
          >
            <Button>1</Button>
            <Button>2</Button>
            <Button>3</Button>
          </ButtonGroup>
        );

        const buttonGroup = getByTestId(testId);
        expect(buttonGroup).toHaveStyleRule('flex-direction', 'row');
        expect(buttonGroup).toHaveStyleRule('justify-content', 'start');
      });
      it('Center: aligns the buttons to the center', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.horizontal}
            alignment={ButtonGroupAlignment.center}
          >
            <Button>1</Button>
            <Button>2</Button>
            <Button>3</Button>
          </ButtonGroup>
        );

        const buttonGroup = getByTestId(testId);
        expect(buttonGroup).toHaveStyleRule('flex-direction', 'row');
        expect(buttonGroup).toHaveStyleRule('justify-content', 'center');
      });
      it('Right: aligns the buttons to the right', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.horizontal}
            alignment={ButtonGroupAlignment.right}
          >
            <Button>1</Button>
            <Button>2</Button>
            <Button>3</Button>
          </ButtonGroup>
        );

        const buttonGroup = getByTestId(testId);
        expect(buttonGroup).toHaveStyleRule('flex-direction', 'row');
        expect(buttonGroup).toHaveStyleRule('justify-content', 'end');
      });
      it('Fill: fills the space with the buttons', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.horizontal}
            alignment={ButtonGroupAlignment.fill}
          >
            <Button>1</Button>
            <Button>2</Button>
            <Button>3</Button>
          </ButtonGroup>
        );

        const buttonGroup = getByTestId(testId);
        expect(buttonGroup).toHaveStyleRule('flex-direction', 'row');
        expect(buttonGroup).toHaveStyleRule('justify-content', 'stretch');
      });
      it('Apart: spreads out the buttons', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.horizontal}
            alignment={ButtonGroupAlignment.apart}
          >
            <Button>1</Button>
            <Button>2</Button>
            <Button>3</Button>
          </ButtonGroup>
        );

        const buttonGroup = getByTestId(testId);
        expect(buttonGroup).toHaveStyleRule('flex-direction', 'row');
        expect(buttonGroup).toHaveStyleRule('justify-content', 'space-between');
      });
    });
    describe('No Space', () => {
      it('Removes the border radius around the buttons', () => {
        const { container, getByTestId } = render(
          <ButtonGroup
            testId={testId}
            noSpace
            orientation={ButtonGroupOrientation.horizontal}
            alignment={ButtonGroupAlignment.left}
          >
            <Button testId={`${testId}-1`}>1</Button>
            <Button testId={`${testId}-2`}>2</Button>
            <Button testId={`${testId}-3`}>3</Button>
          </ButtonGroup>
        );

        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('Vertical', () => {
    describe('Alignment', () => {
      it('Default: aligns the buttons to the left', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
          >
            <Button>1</Button>
            <Button>2</Button>
            <Button>3</Button>
          </ButtonGroup>
        );

        const buttonGroup = getByTestId(testId);
        expect(buttonGroup).toHaveStyleRule('flex-direction', 'column');
        expect(buttonGroup).toHaveStyleRule('justify-content', 'start');
      });
      it('Center: aligns the buttons to the center', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            alignment={ButtonGroupAlignment.center}
          >
            <Button>1</Button>
            <Button>2</Button>
            <Button>3</Button>
          </ButtonGroup>
        );

        const buttonGroup = getByTestId(testId);
        expect(buttonGroup).toHaveStyleRule('flex-direction', 'column');
        expect(buttonGroup).toHaveStyleRule('justify-content', 'center');
      });
      it('Right: aligns the buttons to the right', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            alignment={ButtonGroupAlignment.right}
          >
            <Button>1</Button>
            <Button>2</Button>
            <Button>3</Button>
          </ButtonGroup>
        );

        const buttonGroup = getByTestId(testId);
        expect(buttonGroup).toHaveStyleRule('flex-direction', 'column');
        expect(buttonGroup).toHaveStyleRule('justify-content', 'end');
      });
      it('Fill: fills the space with the buttons', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            alignment={ButtonGroupAlignment.fill}
          >
            <Button>1</Button>
            <Button>2</Button>
            <Button>3</Button>
          </ButtonGroup>
        );

        const buttonGroup = getByTestId(testId);
        expect(buttonGroup).toHaveStyleRule('flex-direction', 'column');
        expect(buttonGroup).toHaveStyleRule('justify-content', 'stretch');
      });
      it('Apart: behaves like the left align', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            alignment={ButtonGroupAlignment.apart}
          >
            <Button>1</Button>
            <Button>2</Button>
            <Button>3</Button>
          </ButtonGroup>
        );

        const buttonGroup = getByTestId(testId);
        expect(buttonGroup).toHaveStyleRule('flex-direction', 'column');
        expect(buttonGroup).toHaveStyleRule('justify-content', 'start');
      });
    });
    describe('No Space', () => {
      it('Does NOT remove the border radius around the buttons', () => {
        const { container, getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            noSpace
          >
            <Button testId={`${testId}-4`}>1</Button>
            <Button testId={`${testId}-5`}>2</Button>
            <Button testId={`${testId}-6`}>3</Button>
          </ButtonGroup>
        );

        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('Props for the buttons', () => {
    describe('Color', () => {
      it('sets the color of the children button components', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            color={ButtonColor.danger}
          >
            <Button testId={`${testId}-1`}>1</Button>
            <Button testId={`${testId}-2`}>2</Button>
            <Button testId={`${testId}-3`}>3</Button>
          </ButtonGroup>
        );

        expect(getByTestId(`${testId}-1`)).toHaveStyleRule('background', magma.colors.danger);
        expect(getByTestId(`${testId}-2`)).toHaveStyleRule('background', magma.colors.danger);
        expect(getByTestId(`${testId}-3`)).toHaveStyleRule('background', magma.colors.danger);
      });
      it('sets the color of the children buttons, except if the button has a color set', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            color={ButtonColor.danger}
          >
            <Button testId={`${testId}-1`}>1</Button>
            <Button testId={`${testId}-2`}>2</Button>
            <Button testId={`${testId}-3`} color={ButtonColor.marketing}>3</Button>
          </ButtonGroup>
        );

        expect(getByTestId(`${testId}-1`)).toHaveStyleRule('background', magma.colors.danger);
        expect(getByTestId(`${testId}-2`)).toHaveStyleRule('background', magma.colors.danger);
        expect(getByTestId(`${testId}-3`)).toHaveStyleRule('background', magma.colors.secondary);
      });
    });

    describe('Size', () => {
      it('sets the size of the children button components', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            size={ButtonSize.large}
          >
            <Button testId={`${testId}-1`}>1</Button>
            <Button testId={`${testId}-2`}>2</Button>
          </ButtonGroup>
        );

        expect(getByTestId(`${testId}-1`)).toHaveStyleRule('height', '56px');
        expect(getByTestId(`${testId}-2`)).toHaveStyleRule('height', '56px');
      });
      it('sets the size of the children buttons, except if the button has a size set', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            size={ButtonSize.large}
          >
            <Button testId={`${testId}-1`}>1</Button>
            <Button testId={`${testId}-2`}>2</Button>
            <Button testId={`${testId}-3`} size={ButtonSize.small}>3</Button>
          </ButtonGroup>
        );

        expect(getByTestId(`${testId}-1`)).toHaveStyleRule('height', '56px');
        expect(getByTestId(`${testId}-2`)).toHaveStyleRule('height', '56px');
        expect(getByTestId(`${testId}-3`)).toHaveStyleRule('height', '28px');
      });
    });

    describe('Variant', () => {
      it('sets the variant of the children button components', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            variant={ButtonVariant.link}
          >
            <Button testId={`${testId}-1`}>1</Button>
            <Button testId={`${testId}-2`}>2</Button>
          </ButtonGroup>
        );

        expect(getByTestId(`${testId}-1`)).toHaveStyleRule('background', 'none');
        expect(getByTestId(`${testId}-2`)).toHaveStyleRule('background', 'none');
      });
      it('sets the variant of the children buttons, except if the button has a variant set', () => {
        const { getByTestId } = render(
          <ButtonGroup
            testId={testId}
            orientation={ButtonGroupOrientation.vertical}
            variant={ButtonVariant.link}
          >
            <Button testId={`${testId}-1`}>1</Button>
            <Button testId={`${testId}-2`}>2</Button>
            <Button testId={`${testId}-3`} variant={ButtonVariant.solid}>3</Button>
          </ButtonGroup>
        );

        expect(getByTestId(`${testId}-1`)).toHaveStyleRule('background', 'none');
        expect(getByTestId(`${testId}-2`)).toHaveStyleRule('background', 'none');
        expect(getByTestId(`${testId}-3`)).toHaveStyleRule('background', magma.colors.primary);
      });
    });
  });
});
