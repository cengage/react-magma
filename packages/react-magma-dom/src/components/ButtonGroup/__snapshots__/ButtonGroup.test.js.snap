// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Horizontal No Space Removes the border radius around the buttons 1`] = `
.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #3942B0;
  border: 0;
  border-color: #3942B0;
  border-radius: 8px;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-family: "Work Sans",Helvetica,sans-serif;
  font-size: 16px;
  font-weight: 500;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 24px;
  margin: 0;
  min-width: 96px;
  overflow: hidden;
  padding: 12px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  touch-action: manipulation;
  -webkit-transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  vertical-align: middle;
  white-space: nowrap;
  width: auto;
}

.emotion-2:not(:disabled):focus {
  outline: 2px solid #0074B7;
  outline-offset: 2px;
}

.emotion-2:not(:disabled):hover,
.emotion-2:not(:disabled):focus {
  background: #2d3489;
  color: #FFFFFF;
}

.emotion-2:not(:disabled):after {
  background: #FFFFFF;
  border-radius: 50%;
  content: '';
  height: 28px;
  left: 50%;
  opacity: 0;
  padding: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) scale(1);
  -ms-transform: translate(-50%,-50%) scale(1);
  transform: translate(-50%,-50%) scale(1);
  -webkit-transition: opacity 1s,-webkit-transform 0.5s;
  -webkit-transition: opacity 1s,transform 0.5s;
  transition: opacity 1s,transform 0.5s;
  width: 28px;
}

.emotion-2:not(:disabled):active {
  background: #202563;
  color: #FFFFFF;
}

.emotion-2:not(:disabled):active:after {
  opacity: 0.4;
  -webkit-transform: translate(-50%,-50%) scale(0);
  -ms-transform: translate(-50%,-50%) scale(0);
  transform: translate(-50%,-50%) scale(0);
  -webkit-transition: -webkit-transform 0s;
  -webkit-transition: transform 0s;
  transition: transform 0s;
}

.emotion-2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-0 {
  visibility: visible;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-12 > button {
  margin: 0;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.emotion-12 > button:first-of-type {
  margin-left: 0;
}

.emotion-12 > button:last-child {
  margin-right: 0;
}

.emotion-12 > button:first-of-type:not(:only-of-type) {
  border-radius: 8px 0 0 8px;
  border-right: 0;
}

.emotion-12 > button:nth-of-type(2) {
  border-left: 1px solid #FFFFFF;
}

.emotion-12 > button:not(:first-of-type) {
  border-radius: 0;
  border-right: 1px solid #FFFFFF;
}

.emotion-12 > button:not(:first-of-type).emotion-12 > button:not(:last-child) {
  border-right: 0;
}

.emotion-12 > button:last-child:not(:only-of-type) {
  border-radius: 0 8px 8px 0;
  border-right: 1px solid #FFFFFF;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #3942B0;
  border: 0;
  border-color: #3942B0;
  border-radius: 8px;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-family: "Work Sans",Helvetica,sans-serif;
  font-size: 16px;
  font-weight: 500;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 24px;
  margin: 0;
  min-width: 96px;
  overflow: hidden;
  padding: 12px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  touch-action: manipulation;
  -webkit-transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  vertical-align: middle;
  white-space: nowrap;
  width: auto;
}

.emotion-2:not(:disabled):focus {
  outline: 2px solid #0074B7;
  outline-offset: 2px;
}

.emotion-2:not(:disabled):hover,
.emotion-2:not(:disabled):focus {
  background: #2d3489;
  color: #FFFFFF;
}

.emotion-2:not(:disabled):after {
  background: #FFFFFF;
  border-radius: 50%;
  content: '';
  height: 28px;
  left: 50%;
  opacity: 0;
  padding: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) scale(1);
  -ms-transform: translate(-50%,-50%) scale(1);
  transform: translate(-50%,-50%) scale(1);
  -webkit-transition: opacity 1s,-webkit-transform 0.5s;
  -webkit-transition: opacity 1s,transform 0.5s;
  transition: opacity 1s,transform 0.5s;
  width: 28px;
}

.emotion-2:not(:disabled):active {
  background: #202563;
  color: #FFFFFF;
}

.emotion-2:not(:disabled):active:after {
  opacity: 0.4;
  -webkit-transform: translate(-50%,-50%) scale(0);
  -ms-transform: translate(-50%,-50%) scale(0);
  transform: translate(-50%,-50%) scale(0);
  -webkit-transition: -webkit-transform 0s;
  -webkit-transition: transform 0s;
  transition: transform 0s;
}

.emotion-2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-0 {
  visibility: visible;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-12 > button {
  margin: 0;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.emotion-12 > button:first-of-type {
  margin-left: 0;
}

.emotion-12 > button:last-child {
  margin-right: 0;
}

.emotion-12 > button:first-of-type:not(:only-of-type) {
  border-radius: 8px 0 0 8px;
  border-right: 0;
}

.emotion-12 > button:nth-of-type(2) {
  border-left: 1px solid #FFFFFF;
}

.emotion-12 > button:not(:first-of-type) {
  border-radius: 0;
  border-right: 1px solid #FFFFFF;
}

.emotion-12 > button:not(:first-of-type).emotion-12 > button:not(:last-child) {
  border-right: 0;
}

.emotion-12 > button:last-child:not(:only-of-type) {
  border-radius: 0 8px 8px 0;
  border-right: 1px solid #FFFFFF;
}

<div>
  <div
    class="emotion-12"
    color="primary"
    data-testid="test-id"
    orientation="horizontal"
  >
    <button
      class="emotion-2 emotion-3"
      color="primary"
      data-testid="test-id-1"
      shape="fill"
      type="button"
    >
      <span
        class="emotion-0 emotion-1"
      >
        1
      </span>
    </button>
    <button
      class="emotion-2 emotion-3"
      color="primary"
      data-testid="test-id-2"
      shape="fill"
      type="button"
    >
      <span
        class="emotion-0 emotion-1"
      >
        2
      </span>
    </button>
    <button
      class="emotion-2 emotion-3"
      color="primary"
      data-testid="test-id-3"
      shape="fill"
      type="button"
    >
      <span
        class="emotion-0 emotion-1"
      >
        3
      </span>
    </button>
  </div>
</div>
`;

exports[`ButtonGroup Vertical No Space Does NOT remove the border radius around the buttons 1`] = `
.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #3942B0;
  border: 0;
  border-color: #3942B0;
  border-radius: 8px;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-family: "Work Sans",Helvetica,sans-serif;
  font-size: 16px;
  font-weight: 500;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 24px;
  margin: 0;
  min-width: 96px;
  overflow: hidden;
  padding: 12px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  touch-action: manipulation;
  -webkit-transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  vertical-align: middle;
  white-space: nowrap;
  width: auto;
}

.emotion-2:not(:disabled):focus {
  outline: 2px solid #0074B7;
  outline-offset: 2px;
}

.emotion-2:not(:disabled):hover,
.emotion-2:not(:disabled):focus {
  background: #2d3489;
  color: #FFFFFF;
}

.emotion-2:not(:disabled):after {
  background: #FFFFFF;
  border-radius: 50%;
  content: '';
  height: 28px;
  left: 50%;
  opacity: 0;
  padding: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) scale(1);
  -ms-transform: translate(-50%,-50%) scale(1);
  transform: translate(-50%,-50%) scale(1);
  -webkit-transition: opacity 1s,-webkit-transform 0.5s;
  -webkit-transition: opacity 1s,transform 0.5s;
  transition: opacity 1s,transform 0.5s;
  width: 28px;
}

.emotion-2:not(:disabled):active {
  background: #202563;
  color: #FFFFFF;
}

.emotion-2:not(:disabled):active:after {
  opacity: 0.4;
  -webkit-transform: translate(-50%,-50%) scale(0);
  -ms-transform: translate(-50%,-50%) scale(0);
  transform: translate(-50%,-50%) scale(0);
  -webkit-transition: -webkit-transform 0s;
  -webkit-transition: transform 0s;
  transition: transform 0s;
}

.emotion-2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-0 {
  visibility: visible;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-12 > button {
  margin: 0;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.emotion-12 > button:first-of-type {
  margin-top: 0;
}

.emotion-12 > button:last-child {
  margin-bottom: 0;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #3942B0;
  border: 0;
  border-color: #3942B0;
  border-radius: 8px;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-family: "Work Sans",Helvetica,sans-serif;
  font-size: 16px;
  font-weight: 500;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 24px;
  margin: 0;
  min-width: 96px;
  overflow: hidden;
  padding: 12px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  touch-action: manipulation;
  -webkit-transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  transition: background 0.35s,border-color 0.35s,box-shadow 0.35s,color 0.35s;
  vertical-align: middle;
  white-space: nowrap;
  width: auto;
}

.emotion-2:not(:disabled):focus {
  outline: 2px solid #0074B7;
  outline-offset: 2px;
}

.emotion-2:not(:disabled):hover,
.emotion-2:not(:disabled):focus {
  background: #2d3489;
  color: #FFFFFF;
}

.emotion-2:not(:disabled):after {
  background: #FFFFFF;
  border-radius: 50%;
  content: '';
  height: 28px;
  left: 50%;
  opacity: 0;
  padding: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) scale(1);
  -ms-transform: translate(-50%,-50%) scale(1);
  transform: translate(-50%,-50%) scale(1);
  -webkit-transition: opacity 1s,-webkit-transform 0.5s;
  -webkit-transition: opacity 1s,transform 0.5s;
  transition: opacity 1s,transform 0.5s;
  width: 28px;
}

.emotion-2:not(:disabled):active {
  background: #202563;
  color: #FFFFFF;
}

.emotion-2:not(:disabled):active:after {
  opacity: 0.4;
  -webkit-transform: translate(-50%,-50%) scale(0);
  -ms-transform: translate(-50%,-50%) scale(0);
  transform: translate(-50%,-50%) scale(0);
  -webkit-transition: -webkit-transform 0s;
  -webkit-transition: transform 0s;
  transition: transform 0s;
}

.emotion-2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-0 {
  visibility: visible;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-12 > button {
  margin: 0;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.emotion-12 > button:first-of-type {
  margin-top: 0;
}

.emotion-12 > button:last-child {
  margin-bottom: 0;
}

<div>
  <div
    class="emotion-12"
    color="primary"
    data-testid="test-id"
    orientation="vertical"
  >
    <button
      class="emotion-2 emotion-3"
      color="primary"
      data-testid="test-id-4"
      shape="fill"
      type="button"
    >
      <span
        class="emotion-0 emotion-1"
      >
        1
      </span>
    </button>
    <button
      class="emotion-2 emotion-3"
      color="primary"
      data-testid="test-id-5"
      shape="fill"
      type="button"
    >
      <span
        class="emotion-0 emotion-1"
      >
        2
      </span>
    </button>
    <button
      class="emotion-2 emotion-3"
      color="primary"
      data-testid="test-id-6"
      shape="fill"
      type="button"
    >
      <span
        class="emotion-0 emotion-1"
      >
        3
      </span>
    </button>
  </div>
</div>
`;
